mmigration/src/transaction.rs

```diff
@@ -65,22 +63,50 @@ impl std::str::FromStr for AppliedMigrationsComment {
     }
 }
 
+pub(crate) struct Transaction<'a> {
+    connection: &'a mut sqlx::postgres::PgConnection,
+    qualified_table_identifier: &'a str,
+    qualified_table_name: &'a crate::QualifiedTableName,
+}
+
 impl Transaction<'_> {
     pub(crate) async fn with_transaction<T, F: AsyncFnMut(&mut Transaction) -> T>(
         client_config: &pg_client::Config,
+        qualified_table_name: &crate::QualifiedTableName,
         mut action: F,
     ) -> T {
         client_config
             .with_sqlx_connection(async |connection| {
+                let qualified_table_identifier =
+                    Self::read_qualified_table_identifier(&mut *connection, qualified_table_name)
+                        .await;
                 Self::begin_serializable_transaction(&mut *connection).await;
-                let mut transaction = Transaction(&mut *connection);
+                let mut transaction = Transaction {
+                    connection: &mut *connection,
+                    qualified_table_identifier: &qualified_table_identifier,
+                    qualified_table_name,
+                };
                 let result = action(&mut transaction).await;
                 Self::commit_transaction(&mut *connection).await;
                 result
             })
             .await
     }
 
+    async fn read_qualified_table_identifier(
+        connection: &mut sqlx::postgres::PgConnection,
+        qualified_table_name: &crate::QualifiedTableName,
+    ) -> String {
+        let row = sqlx::query(r#"SELECT format('%I.%I', $1, $2) table_identifier"#)
```

Comment:
the DB knows best how to escape identifiers. So no need to even pretend do this "right" client side.